src/main/java/org/opentripplanner/netex/validation/ServiceJourneyNonIncreasingPassingTime.java

```diff
@@ -0,0 +1,384 @@
+  }
+
+  /**
+   * A passing time is valid if it is both complete and consistent.
+   */
+  private boolean isValidTimetabledPassingTime(
+    TimetabledPassingTime timetabledPassingTime,
+    ServiceJourneyInfo serviceJourneyInfo
+  ) {
+    if (!hasCompletePassingTime(timetabledPassingTime, serviceJourneyInfo)) {
+      invalidTimetabledPassingTime = timetabledPassingTime;
+      errorType = ErrorType.INCOMPLETE;
+      return false;
+    }
+    if (!hasConsistentPassingTime(timetabledPassingTime, serviceJourneyInfo)) {
+      invalidTimetabledPassingTime = timetabledPassingTime;
+      errorType = ErrorType.INCONSISTENT;
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * A passing time on a regular stop is complete if either arrival or departure time is present. A
+   * passing time on an area stop is complete if both earliest departure time and latest arrival
+   * time are present.
+   */
+  private boolean hasCompletePassingTime(
+    TimetabledPassingTime timetabledPassingTime,
+    ServiceJourneyInfo serviceJourneyInfo
+  ) {
+    if (serviceJourneyInfo.hasRegularStop(timetabledPassingTime)) {
+      return (
+        timetabledPassingTime.getArrivalTime() != null ||
+        timetabledPassingTime.getDepartureTime() != null
+      );
+    }
+    return (
+      timetabledPassingTime.getLatestArrivalTime() != null &&
+      timetabledPassingTime.getEarliestDepartureTime() != null
+    );
+  }
+
+  /**
+   * A passing time on a regular stop is consistent if departure time is after arrival time. A
+   * passing time on an area stop is consistent if latest arrival time is after earliest departure
+   * time.
+   */
+  private boolean hasConsistentPassingTime(
+    TimetabledPassingTime timetabledPassingTime,
+    ServiceJourneyInfo serviceJourneyInfo
+  ) {
+    if (
+      serviceJourneyInfo.hasRegularStop(timetabledPassingTime) &&
+      (
+        timetabledPassingTime.getArrivalTime() == null ||
+        timetabledPassingTime.getDepartureTime() == null
+      )
+    ) {
+      return true;
+    }
+    if (
+      serviceJourneyInfo.hasRegularStop(timetabledPassingTime) &&
+      timetabledPassingTime.getArrivalTime() != null &&
+      timetabledPassingTime.getDepartureTime() != null
+    ) {
+      return (
+        normalizedDepartureTime(timetabledPassingTime) >=
+        normalizedArrivalTime((timetabledPassingTime))
+      );
+    } else {
+      return (
+        normalizedLatestArrivalTime(timetabledPassingTime) >=
+        normalizedEarliestDepartureTime(timetabledPassingTime)
+      );
+    }
+  }
+
+  /**
+   * Regular stop followed by a regular stop: check that arrivalTime(n+1) > departureTime(n)
+   */
+  private boolean isValidRegularStopFollowedByRegularStop(
+    TimetabledPassingTime previousTimetabledPassingTime,
+    TimetabledPassingTime currentTimetabledPassingTime
+  ) {
+    int currentArrivalOrDepartureTime = normalizedArrivalTimeOrElseDepartureTime(
+      currentTimetabledPassingTime
```

Comment:
Would I make sense to create a wrapper class for for `TimetabledPassingTime`, which would be similar to `CallWrapper` for Siri?

That means you can move these helper method in there.

Reply:
If you do, please use the Adaptor design pattern and naming convention. `CallAdaptor` not `CallWrapper`,  
Ref: https://refactoring.guru/design-patterns/adapter

Reply:
Refactored